### PR TITLE
BUG Fix issue with non-asset-admin users encountering errors embedding files

### LINKED ISSUE #730: Insert from files dialog causes errors for non-admin users
As reported from @Worik users with non-admin permissions can sometimes encounter server errors when trying to embed / insert files.

# PRs:

 - [x] https://github.com/silverstripe/silverstripe-assets/pull/115
 - [ ] https://github.com/silverstripe/silverstripe-asset-admin/pull/731

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes #730

Requires https://github.com/silverstripe/silverstripe-assets/pull/115 first

### CODE REVIEW

File: code/Forms/AssetFormFactory.php [hunk 1]
```diff
@@ -56,7 +56,6 @@
 
     public function __construct()
     {
-        $this->constructExtensions();
     }
```
Comment (flamerohr) [inline code/Forms/AssetFormFactory.php]: empty constructor :D

File: tests/php/Forms/FileFormBuilderTest.php [hunk 2]
```diff
@@ -211,11 +211,29 @@ public function testInsertImageForm()
...
+    public function testInsertImageFormDenied()
+    {
+        $this->logInWithPermission('SOME_PERMISSION');
+
+        $image = $this->objFromFixture(Image::class, 'image1');
+        $controller = new AssetAdmin();
+        $builder = new ImageFormFactory();
+        $form = $builder->getForm($controller, 'EditForm', ['Record' => $image, 'Type' => FileFormFactory::TYPE_INSERT_MEDIA]);
+
+        // Ensure "insert" button exists
```
Comment (flamerohr) [inline tests/php/Forms/FileFormBuilderTest.php]: incorrect comment - "does not exist" is more accurate

File: code/Forms/AssetFormFactory.php (hunk 1 above)
Comment (flamerohr) [reply code/Forms/AssetFormFactory.php]: hmm, took a bit of searching to find a reference to this, but ok, leave this in

Comment (robbieaverill) [reply code/Forms/AssetFormFactory.php]: Heh - you learn something every day =)

File: tests/php/Forms/FileFormBuilderTest.php (hunk 2 above)
Comment (tractorcow) [reply tests/php/Forms/FileFormBuilderTest.php]: Fixed.

**Comment by tractorcow** [comment]
I fixed everything I wanted to and pushed.

**Pull request merged**